edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.CloudProxy/DeviceScopeApiClient.cs

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Hub.CloudProxy
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Net;
+    using System.Net.Http;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Client.Exceptions;
+    using Microsoft.Azure.Devices.Common;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Azure.Devices.Edge.Util.TransientFaultHandling;
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
+
+    public class DeviceScopeApiClient : IDeviceScopeApiClient
+    {
+        const int RetryCount = 3;
+        static readonly ITransientErrorDetectionStrategy TransientErrorDetectionStrategy = new ErrorDetectionStrategy();
+        static readonly RetryStrategy TransientRetryStrategy =
```

Comment:
We should probably take the `RetryStrategy` through a constructor and have a reasonable default in the current constructor.

Reply:
Sure